packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -292,12 +292,10 @@ export function initInput(
     typeof checkedOrDefault !== 'symbol' &&
     !!checkedOrDefault;
 
-  // The checked property never gets assigned. It must be manually set.
-  // We don't want to do this when hydrating so that existing user input isn't
-  // modified
-  // TODO: I'm pretty sure this is a bug because initialValueTracking won't be
-  // correct for the hydration case then.
-  if (!isHydrating) {
+  if (isHydrating) {
```

Comment:
tempted to change this to `isHydrating && checked == null`, so you can interactions before hydration work for uncontrolled but make sure they get lost for controlled so we're in sync

Reply:
ok I did it. this seems like the most sensible behavior to me without going as far as firing onChange during hydration

Reply:
no, I take it back. per

https://github.com/facebook/react/blob/34de2986dfeddda31cb76d298c6d59b271d976fa/packages/react-dom/src/__tests__/ReactDOMServerIntegrationUserInteraction-test.js#L247

we leave user interactions on `input` value and `select` despite DOM and React being out of sync so I guess we should for radio and checkbox too.

please review this PR as I had it originally. thanks <3